test/libsolidity/syntaxTests/functionTypes/declaration_type_conversion.sol

```diff
@@ -0,0 +1,11 @@
+contract D {
+	function f() external {}
+	function g() external {}
+}
+contract C {
+	function f(bool c) public pure {
+		(c ? D.f : D.g);
+	}
+}
+// ----
+// TypeError 1080: (117-130): True expression's type function D.f() does not match false expression's type function D.g().
```

Comment:
It seems that the `Type::commonType` ignores the error message returned by `isImplicitlyConvertibleTo` and just returns a `nullptr` to `TypeChecker::visit`. Shouldn't the error message be the `BoolResult::err` defined at line [3134](https://github.com/ethereum/solidity/pull/14365/files#diff-5cc133bfc337e9ca7fe7ee9842d5ddfa492fa559149cf8acec4a7d9ede50c9aaR3134)

Reply:
Maybe, yes - the error reporting across those has always been broken - we started years ago to introduce the ``Result`` thing, but never went through with it properly in the end.
But if at all, it's something to be fixed separately: It's independent of this PR and the error message is actually just fine as is as well.